### PR TITLE
[#4988] fix(doris-catalog): Fix the missing distribution information when loading Doris tables

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Loading distribution information when obtaining Doris tables.

### Why are the changes needed?

It's a bug to be fixed.

Fix: #4988 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

IT.

### CODE REVIEW

**Comment by yuqi1129** [comment]
This PR is not ready for review until #4985 is merged.

File: catalogs/catalog-jdbc-doris/src/test/java/org/apache/gravitino/catalog/doris/operation/TestDorisTableOperations.java [hunk 1]
```diff
@@ -94,6 +95,54 @@ private static Map<String, String> createProperties() {
     return properties;
   }
 
+  @Test
```
Comment (mchades) [inline catalogs/catalog-jdbc-doris/src/test/java/org/apache/gravitino/catalog/doris/operation/TestDorisTableOperations.java:98]: plz also add relative IT

File: catalogs/catalog-jdbc-common/src/main/java/org/apache/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 2]
```diff
@@ -204,6 +205,10 @@ public JdbcTable load(String databaseName, String tableName) throws NoSuchTableE
       Transform[] tablePartitioning = getTablePartitioning(connection, databaseName, tableName);
       jdbcTableBuilder.withPartitioning(tablePartitioning);
 
+      // 5.Get distribution information
+      Distribution distribution = getDistributionInfo(connection, databaseName, tableName);
+      jdbcTableBuilder.withDistribution(distribution);
+
       // 5.Get table properties
```
Comment (mchades) [inline catalogs/catalog-jdbc-common/src/main/java/org/apache/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:212]: this should be `6.xxx` ?

**Review by mchades — COMMENTED** [review]
The docs also need to be update

**Review by mchades — COMMENTED** [review]
`docs/jdbc-doris-catalog.md` this doc needs to be updated

File: catalogs/catalog-jdbc-doris/src/main/java/org/apache/gravitino/catalog/doris/operation/DorisTableOperations.java [hunk 3]
```diff
@@ -221,7 +221,7 @@ private static void validateDistribution(Distribution distribution, JdbcColumn[]
 
     Preconditions.checkArgument(
         Strategy.HASH == distribution.strategy() || Strategy.EVEN == distribution.strategy(),
-        "Doris only supports HASH or EVEN distribution strategy");
+        "Doris only supports HASH or EVEN(RANDOM) distribution strategy");
```
Comment (mchades) [inline catalogs/catalog-jdbc-doris/src/main/java/org/apache/gravitino/catalog/doris/operation/DorisTableOperations.java:224]: if the `Strategy` is `EVEN`, we should check the `distribution.expressions()` must be empty?

File: docs/jdbc-doris-catalog.md [hunk 4]
```diff
@@ -161,6 +161,15 @@ Note that although Gravitino supports several partitioning strategies, Apache Do
 The `fieldName` specified in the partitioning attributes must be the name of columns defined in the table.
 :::
 
+### Table distribution
+
+Users can also specify the distribution strategy when creating tables in the Doris catalog. Currently, the Doris catalog supports the following distribution strategies:
+- `HASH`
+- `RANDOM`
+
+For the `RANDOM` distribution strategy, Gravitino uses the `EVEN` to represent it. More information about the distribution strategy defined in Gravitino can be found [here](./table-partitioning-bucketing-sort-order-indexes.md#table-distribution).
```
Comment (mchades) [inline docs/jdbc-doris-catalog.md:170]: the file link `./table-partitioning-bucketing-sort-order-indexes.md#table-distribution` is incorrect, the term `bucketing` in file name should be replaced with `distribution`

Comment (yuqi1129) [reply docs/jdbc-doris-catalog.md:170]: Sorry, Forget to rename the file when I change the content there.

**Review by mchades — APPROVED** [review]
LGTM.  Do you want to review it? @FANNG1

File: catalogs/catalog-jdbc-common/src/main/java/org/apache/gravitino/catalog/jdbc/operation/JdbcTableOperations.java [hunk 5]
```diff
@@ -236,6 +241,20 @@ protected Transform[] getTablePartitioning(
...
   }
 
+  /**
+   * Get the distribution information of the table, including the distribution type and the fields
+   *
+   * @param connection
+   * @param databaseName
+   * @param tableName
+   * @return
+   * @throws SQLException
```
Comment (mchades) [inline catalogs/catalog-jdbc-common/src/main/java/org/apache/gravitino/catalog/jdbc/operation/JdbcTableOperations.java:251]: missing comments

**Review by mchades — CHANGES_REQUESTED** [review]

**Review by mchades — APPROVED** [review]

**Pull request merged**